src/pybind/mgr/cephadm/tests/test_services.py

```diff
@@ -1267,46 +1287,47 @@ def fake_get_addr(hostname: str) -> str:
         # for the host1 nfs daemon as we'll end up giving that
         # one higher rank_generation but the same rank as the one
         # on host2
+        haproxy_txt = (
+            '# This file is generated by cephadm.\n'
+            'global\n'
+            '    log         127.0.0.1 local2\n'
+            '    chroot      /var/lib/haproxy\n'
```

Comment:
Locally, this instruction seems to fail for me with not much in the logs besides: `Cannot chroot /var/lib/haproxy`. When I comment this out, i can deploy the ingress service fine - but that doesn't seem right... I'm using quay.io/ceph/haproxy:2.3 as the image.

Reply:
The lines in question that you tagged are part of the unit tests. This should not be directly relevant to the code running on a live cluster. I'm guessing the line in the [haproxy template file](https://github.com/ceph/ceph/pull/50614/files/b331d875d285476f8b8ed269d97002afd4d15598#diff-32a0e1886eacaacda4c6a8277c18af9f13634ef5dd0c8e779bf8a49c9b183ff3R4) is  really what you meant to highlight?
If so, I'm unclear why it would behave differently for you than other cephadm deployments of haproxy... this PR doesn't change that part of the config AFAICT.  Are you deploying the container in a (partially) manual way?

Reply:
Interesting. I tested with images built using the nfs ganesha rpms built by shaman (https://shaman.ceph.com/builds/nfs-ganesha/) but I don't remember exactly which (I think ceph main tag from https://quay.ceph.io/repository/ceph-ci/ceph - but it long enough ago that I forget if it was for sure.

I did not have the problem you encountered. Let's wait for at least one theuthology run of this current series of patches. If it passes the issue may indeed be related to how your image(s) are built, and filing an issue later would be fine.